aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SubscriptionProcessor.java

```diff
@@ -224,7 +224,19 @@ void startDrainingMutationBuffer(Action onPipelineBroken) {
                 .doOnSubscribe(disposable ->
                     LOG.info("Starting processing subscription data buffer.")
                 )
-                .flatMapCompletable(mutation -> merger.merge(mutation.modelWithMetadata()))
+                .flatMapCompletable(mutation -> {
+                    ModelWithMetadata<? extends Model> original = mutation.modelWithMetadata();
+                    if (original.getModel() instanceof SerializedModel) {
+                        SerializedModel originalModel = (SerializedModel) original.getModel();
+                        SerializedModel newModel = SerializedModel.builder()
+                            .serializedData(originalModel.getSerializedData())
+                            .modelSchema(mutation.modelSchema())
```

Comment:
The response from AppSync doesn't contain any information about what _type_ it is. Previously, we used to know this because of the `<T>`. Now, I have to hack the `modelSchema` back in at this higher level. The `SubscriptionProcessor` knows which subscription the data came back on, so it can append the schema, here.

Reply:
Just a clarification...are the terms new and original referring to the type of model you're working with (i.e. SerializedModel vs POJO) or is it referring to the data (i.e. original = existing data and new = updated data)?

Reply:
`originalModel` means "before I patch this `SerializedModel` instance's `modelSchema` field," and `newModel` means "the `SerializedModel` we got back from the sync command, but with a patched-up `modelSchema` field."

Reply:
Not a blocker for this PR, but it would be nice to see a `newBuilder()` method on the `SerializedModel` class, so you could simplify this to: 

```
SerializedModel newModel = originalModel.newBuilder()
    .modelSchema(mutation.modelSchema())
    .build();
```